### ISSUE TITLE
Not working on Ubuntu Gnome 16.10

### ISSUE THREAD

**Issue by RicardoEPRodrigues** [bharadwaj-raju/WeatherDesk#29]
To put it in simple terms it crashes.

In more detail, first it didn't have a needed dependency for the file _psutil_, that I solved by running `sudo apt install python3-psutil`.

But now it crashes with the following:

    [City checking] No error.
    [Main loop] 
    Traceback (most recent call last):
      File "WeatherDesk.py", line 396, in <module>
        weather = str(weather_json['query']['results']['channel']['item']['condition']['text']).lower()
    TypeError: 'NoneType' object is not subscriptable

**Comment by bharadwaj-raju** [comment]
Yes, *sometimes* that error occurs. It's a problem on the API's end, nothing we can do.

Try a few test runs with `--one-time-run`.

**Comment by RicardoEPRodrigues** [comment]
`WeatherDesk.py: error: unrecognized arguments: --one-time-run`

What am I doing wrong? xD

Note: the program doesn't crash, but it also doesn't change the background...

**Comment by bharadwaj-raju** [comment]
> What am I doing wrong? xD

Sorry, my bad. The `--one-time-run` is a new feature that is as-yet not pushed to GitHub (but it will be in a few minutes).

Then just download the latest and run it again.

**Comment by bharadwaj-raju** [comment]
Now I've updated it, please try again with latest.

**Comment by RicardoEPRodrigues** [comment]
Updated and it ran it as you requested.

I noted inconsistent output. Sometimes it works, sometimes it doesn't...

When it doesn't it shows the error previously mentioned.

**Comment by bharadwaj-raju** [comment]
Can you try visiting this URL in your browser?

```
https://query.yahooapis.com/v1/public/yql?q=select%20%2A%20from%20geo.places(5)%20where%20text%3D"CITY"&format=json
```

Replacing `CITY` for your city.

**Comment by RicardoEPRodrigues** [comment]
I always receive the same result:
[result.txt](https://github.com/bharadwaj-raju/WeatherDesk/files/750903/result.txt)

No error from that link.

**Comment by bharadwaj-raju** [comment]
For a possible solution, try increasing `--wait` from the default of `600` seconds.

**Comment by RicardoEPRodrigues** [comment]
I tried to increase the wait time to 2000 and ran a few `--one-time-run`, but it's still inconsistent.

**Comment by RicardoEPRodrigues** [comment]
Let me just say, it the errors doesn't break the app and just means that one update failed, then I won't worry and just use it as is.

**Comment by bharadwaj-raju** [comment]
I, too, would really like to fix this but cannot as it happens on what seems to be a random basis.

**Comment by RicardoEPRodrigues** [comment]
I don't know if I should close the issue, since it is not resolved... Maybe you should tag it as a WontFix and close it.

**Comment by bharadwaj-raju** [comment]
Should I close this issue?
